src/context.ts

```diff
@@ -151,8 +151,7 @@ export class Context implements Required<ContextValue> {
     throw new SdkError("subgraph configuration must be defined in Context for this feature to work.");
   }
 
-  get partnerId(): string {
-    if (typeof this.ctx.partnerId !== "undefined") return this.ctx.partnerId;
-    throw new SdkError("partnerId must be defined in Context for this feature to work.");
+  get partnerId(): string | undefined {
```

Comment:
Didn't want to do the same for the other values as then they'd become optional/a wider refactor